contrib/utilities/run_clang_tidy.sh

```diff
@@ -72,4 +56,15 @@ CC=clang CXX=clang++ cmake $ARGS "$SRC" || (echo "cmake failed!"; false) || exit
 
 cmake --build . --target expand_all_instantiations || (echo "make expand_all_instantiations failed!"; false) || exit 3
 
-run-clang-tidy.py -p . -checks="$CHECKS" -quiet -header-filter="$SRC/include/*" -fix
+# finally run it:
+# pipe away stderr (just contains nonsensical "x warnings generated")
+# pipe output to output.txt
+run-clang-tidy.py -p . -quiet -header-filter="$SRC/include/*" 2>error.txt >output.txt
+if [[ -n $(grep "warning: " output.txt) ]] || [[ -n $(grep "error: " output.txt) ]]; then
```

Comment:
ShellCheck says:
```
     ^-- SC2143: Use grep -q instead of comparing output with [ -n .. ].
                                                 ^-- SC2143: Use grep -q instead of comparing output with [ -n .. ]. 
```